client/errors.go

```diff
@@ -8,6 +8,11 @@ import (
 // ErrConnectionFailed is an error raised when the connection between the client and the server failed.
 var ErrConnectionFailed = errors.New("Cannot connect to the Docker daemon. Is the docker daemon running on this host?")
 
+// PermissionDenied returns an error with socketFile in the error message when a unix socket can not be opened.
+func PermissionDenied(socketFile string) error {
+	return fmt.Errorf("Cannot open %s. Permission Denied.", socketFile)
```

Comment:
Can we just define an error like we have for connection failed and use `errors.Wrap` (from github.com/pkg/errors) to provide the extra context?